### PR TITLE
[MSEARCH-812] Increase the limit to fetch more than 1k locations

### PR DESCRIPTION
**Pull request by azizbekxm**
### Purpose
In scope of ticket we need to fetch all available locations to check by code https://github.com/folio-org/mod-gobi/pull/238
### Approach
Increasing max limit for api search/consortium/locations
https://folio-org.atlassian.net/browse/MSEARCH-812

### CODE REVIEW

**Review by SerhiiNosko — APPROVED** [review]

File: src/main/resources/swagger.api/parameters/consortium-locations-limit-param.yaml [hunk 1]
```diff
@@ -4,5 +4,5 @@ description: Limit the number of elements returned in the response.
 schema:
   type: integer
   minimum: 0
-  maximum: 1000
+  maximum: 10000
```
Comment (mukhiddin-yusuf) [inline src/main/resources/swagger.api/parameters/consortium-locations-limit-param.yaml:7]: keep in mind that this API uses pagination search and opensearch/elasticsearch puts a limit of 10.000 search results for ```from + size``` (or ```offset + limit```)
https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html
https://opensearch.org/docs/latest/search-plugins/searching-data/paginate/

now, if API is called with max limit then offset must be 0, or if the use of using this API with ```offset + limit > 10.000``` is possible or quite frequent then search query needs to be changed

Comment (SerhiiNosko) [reply src/main/resources/swagger.api/parameters/consortium-locations-limit-param.yaml:7]: we will use this without limit and offset param, fetching only 1000 can not be enought, but 10k is good. So 10000 as ES suggests as offset + limit

Comment (psmagin) [reply src/main/resources/swagger.api/parameters/consortium-locations-limit-param.yaml:7]: I suggest to add validation in code to check that offset + limit is less then 10000

**Comment by psmagin** [comment]
Will the story https://folio-org.atlassian.net/browse/MSEARCH-809 be closed after this PR merge? If no, then I suggest to split it

File: src/main/resources/swagger.api/parameters/consortium-locations-limit-param.yaml (hunk 1 above)
Comment (SerhiiNosko) [reply src/main/resources/swagger.api/parameters/consortium-locations-limit-param.yaml:7]: added AC regarding this into newly created ticket: https://folio-org.atlassian.net/browse/MSEARCH-812

**Comment by SerhiiNosko** [comment]
> Will the story https://folio-org.atlassian.net/browse/MSEARCH-809 be closed after this PR merge? If no, then I suggest to split it

Extracted separate story for this: https://folio-org.atlassian.net/browse/MSEARCH-812
By the way, we checked that /search/consortium/locations is working also in non-ECS mode so we are going to use it in both ECS and non-ECS for mod-gobi to avoid an additional check if ECS mode is enabled. If you see some issues in this approach please let us know.

File: src/main/resources/swagger.api/parameters/consortium-locations-limit-param.yaml (hunk 1 above)
Comment (azizbekxm) [reply src/main/resources/swagger.api/parameters/consortium-locations-limit-param.yaml:7]: Added validation for this case

File: src/main/java/org/folio/search/service/consortium/ConsortiumLocationService.java [hunk 2]
```diff
@@ -41,5 +43,11 @@ private void validateSortByValue(String sortBy) {
     }
   }
 
+  private void validatePaginationParameters(Integer limit, Integer offset) {
+    if (limit + offset > MAX_RESULT_WINDOW) {
+      throw new IllegalArgumentException("The combination of limit and offset exceeds "
+        + "the maximum result window of" + MAX_RESULT_WINDOW);
```
Comment (psmagin) [inline src/main/java/org/folio/search/service/consortium/ConsortiumLocationService.java:49]:
```suggestion
        + "the maximum result window of " + MAX_RESULT_WINDOW);
```

File: src/test/java/org/folio/search/service/consortium/ConsortiumLocationServiceTest.java [hunk 3]
```diff
@@ -59,6 +60,29 @@ void fetchLocations_ValidSortBy() {
     verify(executor).execute(eq(tenantId), any(Supplier.class));
   }
 
+
+  @Test
+  void fetchCampuses_InvalidSortBy() {
```
Comment (psmagin) [inline src/test/java/org/folio/search/service/consortium/ConsortiumLocationServiceTest.java:65]:
```suggestion
  void fetchLocations_InvalidSortBy() {
```

File: src/test/java/org/folio/search/service/consortium/ConsortiumLocationServiceTest.java [hunk 4]
```diff
@@ -59,6 +60,29 @@ void fetchLocations_ValidSortBy() {
...
+    var limit = 10;
+    var offset = 0;
+
+    Assertions.assertThrows(IllegalArgumentException.class, () ->
+      service.fetchLocations(CONSORTIUM_TENANT, CONSORTIUM_TENANT, limit, offset, "invalid", sortOrder)
+    );
+  }
+
+  @Test
+  void fetchCampuses_InvalidPaginationParameters() {
```
Comment (psmagin) [inline src/test/java/org/folio/search/service/consortium/ConsortiumLocationServiceTest.java:76]:
```suggestion
  void fetchLocations_InvalidPaginationParameters() {
```

File: src/main/java/org/folio/search/service/consortium/ConsortiumLocationService.java (hunk 2 above)
Comment (azizbekxm) [reply src/main/java/org/folio/search/service/consortium/ConsortiumLocationService.java]: thanks fixed

File: src/test/java/org/folio/search/service/consortium/ConsortiumLocationServiceTest.java (hunk 3 above)
Comment (azizbekxm) [reply src/test/java/org/folio/search/service/consortium/ConsortiumLocationServiceTest.java]: thanks, fixed

File: src/test/java/org/folio/search/service/consortium/ConsortiumLocationServiceTest.java (hunk 4 above)
Comment (azizbekxm) [reply src/test/java/org/folio/search/service/consortium/ConsortiumLocationServiceTest.java]: fixed as well

**Review by mukhiddin-yusuf — APPROVED** [review]

**Pull request merged**